graph.py

```diff
@@ -60,3 +60,42 @@ def adjacent(self, n1, n2):
             return True
         else:
             return False
+
+    def breadth_first_traversal(self, start):
+        visited = set()
+        queue = [start]
+        return_value = []
+        while queue:
+            node = queue.pop(0)
+            if node not in visited:
+                visited.add(node)
+                return_value.append(node)
+                queue.extend(self.dict_graph[node])
+        return return_value
+
+    def _depth_first_visitor(self, node, visited, return_value):
+        if node in visited:
+            return
+
+        visited.add(node)
+        return_value.append(node)
+        for child_node in self.dict_graph[node]:
+            self._depth_first_visitor(child_node, visited, return_value)
```

Comment:
This method operates by "side effect", meaning that it mutates an object in-place rather than returning a value.  It's is tempting to do this, but it can lead to problems tracking where changes to your data actually happened.  It might be better to return the mutated `return_value` so that you can re-assign it outside this call.  That might be more clear to read, and will be a bit easier to debug in the long run.